packages/apps/src/configs/chains/chain-utils.ts

```diff
@@ -0,0 +1,63 @@
+import { ChainType, InternalChainId } from '@webb-dapp/apps/configs';
+
+export const byteArrayToNum = (arr: number[]): number => {
+  let n = 0;
+  for (let i of arr) n = n * 256 + i;
+  return n;
+};
+
+/**
+ * @param num the number to be converted
+ * @param min the minimum bytes the array should hold (in the case of requiring empty bytes to match rust values)
+ * @returns
+ */
+export const numToByteArray = (num: number, min: number): number[] => {
+  let arr = [];
+  while (num > 0) {
+    arr.push(num % 256);
+    num = Math.floor(num / 256);
+  }
+  arr.reverse();
+  // maintain minimum number of bytes
+  while (arr.length < min) {
+    arr = [0, ...arr];
+  }
+  return arr;
+};
+
+export const computeChainIdType = (chainType: ChainType, chainId: number | InternalChainId): number => {
+  let chainTypeArray = numToByteArray(chainType, 2);
+  let chainIdArray = numToByteArray(chainId, 4);
+  let fullArray = [...chainTypeArray, ...chainIdArray];
+  return byteArrayToNum(fullArray);
+};
+
+// ! ChainId should NOT be InternalChainId
+export interface ChainTypeId {
+  chainType: ChainType;
+  chainId: number;
```

Comment:
Can we give a type to chain id? , and reuse the `ChainTypeId` type everywhere we have the combo (chainType ,chainId) instead of passing them sapretly (ex here https://github.com/webb-tools/webb-dapp/blob/711d7653366b3e4cfa7c08518985fff3c63e0a63/packages/bridge/src/components/DepositConfirm/DepositConfirm.tsx#L138-L138 )

```suggestion
  chainId: InternalChainId;
```